### PR TITLE
add ddo crud

### PR DESCRIPTION
**Pull request by an7e**
PR for #48 

Changes proposed in this PR:

- added schemas for ddo, nonce, indexer
- added Database main class to initialize collections and provide interaction with them
- added DdoDatabase, NonceDatabase and IndexerDatabase classes to provide CRUD methods for ddo, nonce, indexer

### CODE REVIEW

File: src/@types/OceanNode.ts [hunk 1]
```diff
@@ -3,13 +3,11 @@ import { OceanProvider } from '../components/Provider/index'
...
 import { Stream } from 'stream'
+import { TypesenseConfigOptions } from './Typesense'
 import { Blockchain } from '../utils/blockchain'
 
 export interface OceanNodeDBConfig {
-  host: string
-  user: string
-  pwd: string
-  dbname: string
+  typesense: TypesenseConfigOptions
```
[the remark this answers is not in the document]
Comment (an7e) [reply src/@types/OceanNode.ts]: @alexcos20 changed to URL, need your feedback

File: src/utils/config.ts [hunk 2]
```diff
@@ -96,10 +96,16 @@ export async function getConfig(): Promise<OceanNodeConfig> {
     hasProvider: true,
     httpPort: getIntEnvValue(process.env.HTTP_API_PORT, 8000),
     dbConfig: {
-      dbname: 'oceannode',
-      host: '127.0.0.1',
-      user: 'oceannode',
-      pwd: 'oceannode'
+      typesense: {
```
[the remark this answers is not in the document]
Comment (an7e) [reply src/utils/config.ts]: @alexcos20 changed to URL, need your feedback

File: jest.config.js [hunk 3]
```diff
@@ -1,4 +1,17 @@
 export default {
   preset: 'ts-jest',
-  testEnvironment: 'node',
+  extensionsToTreatAsEsm: ['.ts'],
```
Comment (bogdanfazakas) [inline jest.config.js:3]: do we still need this? or is it better to rebase develop and keep only mocha ?

Comment (paulo-ocean) [reply jest.config.js]: yeah, i forgot to delete that as well on develop... we don't need it. just kill it :-)

**Review by alexcos20 — APPROVED** [review]
lgtm, let's merge, we can always solve issues later if we need to

**Pull request merged**